.github/workflows/ci.yaml

```diff
@@ -74,44 +75,45 @@ jobs:
         id: prep
         run: |
           hack/build/ci/prepare-build-variables.sh
-      - name: Docker metadata
-        uses: docker/metadata-action@369eb591f429131d6889c46b94e711f089e6ca96 # v5.6.1
-        id: meta
-        with:
-          images: ${{ env.DOCKER_REPOSITORY }}
-          tags: ${{ steps.prep.outputs.docker_image_tag }}
-          labels: |
-            ${{ steps.prep.outputs.docker_image_labels }}
-            vcs-ref=${{ github.sha }}
-      - name: Prepare build parameters
-        id: prepenv
-        run: |
-          # Set output parameters.
-          # Reason: global envs do not work in workflow calls
-          # More info: https://github.com/actions/runner/issues/480#issuecomment-1021278915
-
-          echo "registry=${{ env.DOCKER_REGISTRY }}" >> "$GITHUB_OUTPUT"
-          echo "repository=${{ env.DOCKER_REPOSITORY }}" >> "$GITHUB_OUTPUT"
     outputs:
-      labels: ${{ steps.meta.outputs.labels }}
+      labels: |
+        ${{ steps.prep.outputs.docker_image_labels }}
+        vcs-ref=${{ github.sha }}
       version: ${{ steps.prep.outputs.docker_image_tag }}
-      registry: ${{ steps.prepenv.outputs.registry }}
-      repository: ${{ steps.prepenv.outputs.repository }}
+      go_linker_args: ${{ steps.prep.outputs.go_linker_args }}
 
-  build:
-    name: Build images
+  build-push:
+    name: Build and push images
     runs-on: ubuntu-latest
-    needs: [prepare, tests, linting]
-    strategy:
-      matrix:
-        platform: [amd64, arm64]
+    needs: [prepare]
+    if: ${{ !github.event.pull_request.head.repo.fork }}
     steps:
       - name: Checkout
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
-      - name: Build image
-        if: matrix.platform == 'amd64' || github.ref_protected
-        uses: ./.github/actions/build-image
+      - name: Set up Golang
```

Comment:
Out of curiosity, why do we need `set-up go` as a precondition to building an image using Podman?

Reply:
mainly want to have it so we build with the same version as in the go.mod
```
go-version-file: "${{ github.workspace }}/go.mod"
```

Reply:
I still don't get it, why you need `go.mod` for building image using podman.

Reply:
truuuue

Reply:
inside image we use go.mod, but why we need setupgo outside? am I miss something?

Reply:
that was what I thought too.